pkg/reconciler/ingress/resources/gateway.go

```diff
@@ -189,6 +196,16 @@ func makeWildcardGateways(ctx context.Context, originWildcardSecrets map[string]
 	return gateways, nil
 }
 
+// IsDNS1123Label tests for a string that conforms to the definition of a label in
+// DNS (RFC 1123).
+// This function is copied from https://github.com/istio/istio/blob/806fb24bc121bf93ea06f6a38b7ccb3d78d1f326/pkg/config/labels/instance.go#L97
+// We directly copy this function instead of importing it into vendor and using it because
+// if this function is changed in the upstream (for example, Istio allows the dot in the future), we don't want to
+// import the change without awareness because it could break the compatibility of Gateway name generation.
+func isDNS1123Label(value string) bool {
```

Comment:
Can we use the existing https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L187 ?

Reply:
I think technically we can either use istio one or k8s one.

But I am inclined to copy the function into this repo because if there is any change to this function in the upstream (e.g. allowing dot in the future), then the gateway name could be changed implicitly and we may not be aware of this.
If the version of the function we use in the our code is not compatible with the requirement of Istio or k8s in the future, then our E2E tests should be failed and we will get notice about it. WDYT?

Reply:
I get your point, but this is not some random function, it follows RFC 1123. It is not going to change.
Up to you.